### PR TITLE
ADIOS2: New version 2.9.2

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
I do not understand the package compatibility error. (I also asked in the Julia Slack workspace in the `#binarybuilder` channel.)

**Comment by giordano** [comment]
https://github.com/JuliaPackaging/Yggdrasil/pull/7626#issuecomment-1813258097

**Comment by giordano** [comment]
Uhm, blosc2 can't be found https://buildkite.com/julialang/yggdrasil/builds/6667#018bda1e-c5f4-4e5f-807e-fb405bc27e6b/612-1181
```
[21:53:38] -- Configuring done
[21:53:38] CMake Error at source/adios2/CMakeLists.txt:6 (add_library):
[21:53:38]   Target "adios2_core" links to target "Blosc2::blosc2_static" but the target
[21:53:38]   was not found.  Perhaps a find_package() call is missing for an IMPORTED
[21:53:38]   target, or an ALIAS target is missing?
[21:53:38] 
[21:53:38] 
```
but it's looking for the static library?

**Comment by eschnett** [comment]
It's these lines in ADIOS2:
```
  set(adios2_blosc2_tgt Blosc2::Blosc2)
  if (Blosc2_VERSION VERSION_GREATER_EQUAL 2.10.1)
    if (Blosc2_shlib_available)
      set(adios2_blosc2_tgt Blosc2::blosc2_$<IF:$<BOOL:${ADIOS2_Blosc2_PREFER_SHARED}>,shared,static>)
    else()
      set(adios2_blosc2_tgt Blosc2::blosc2_static)
    endif()
  endif()
```
They check whether Blosc2 has shared libraries available, but don't check for the static ones...

**Comment by giordano** [comment]
Thanks for looking into it! Worth reporting it upstream?

File: A/ADIOS2/platforms.md
Comment (giordano) [inline A/ADIOS2/platforms.md:1]: This file doesn't look necessary?

Comment (eschnett) [reply A/ADIOS2/platforms.md:1]: bah. thanks.

**Comment by eschnett** [comment]
Yes, I reported 3 issues upstream. See https://github.com/ornladios/ADIOS2/issues/3924.

**Comment by giordano** [comment]
So, we either compile against OpenMPI 4 (which supported also FreeBSD), or remove FreeBSD when linking to OpenMPI.

**Pull request merged**